src/catch2/internal/catch_clara.hpp

```diff
@@ -448,16 +469,21 @@ namespace Catch {
                     m_ref( ref ) {}
 
             public:
-                template <typename T>
+                template <typename T, typename = typename std::enable_if<!Detail::is_unary_function<T>::value>::type>
```

Comment:
```suggestion
                template <typename T, typename = typename std::enable_if_t<!Detail::is_unary_function<T>::value>>
```

v3 targets C++14 as baseline, so traits can be used with the `_t` suffix. (Sadly, no `_v` for values)